tests/pgautofailover_utils.py

```diff
@@ -358,10 +364,21 @@ def create(self):
                         '--pgdata', self.datadir,
                         '--pgport', str(self.port),
                         '--nodename', self.nodename]
+        
+        if self.auth_method is not None:
+            init_command.extend(['--auth', self.auth_method])    
+        
         init_proc = self.vnode.run(init_command)
         wait_or_timeout_proc(init_proc,
                              name="create monitor",
                              timeout=COMMAND_TIMEOUT)
+        if self.password is not None:
+            alter_user_set_passwd_command = 'alter user ' + self.username + ' with password' + ' \'' + self.password + '\''
```

Comment:
Please use Python string building facilities here:

``` python
    alter_user_set_passwd_command = "alter user %s with password \"%s\"" % (self.username, self.password)
```

Reply:
I think we have 4 different Postgres users that we use at different points:

  1. the `--username`, `docker` in the test environment (OS logged in user)
  2. `autoctl`, the owner of the `pg_auto_failover` database
  3. `autoctl_node`, used by pg_autoctl in the API to the monitor, as per `pg_autoctl show uri`, or the one used in the `pg_autoctl create Postgres --monitor` option
  4. `pgautofailover_monitor` when the monitor connects to the Postgres nodes for health checks

Which user are we setting a password for in that code path? and which user are we providing a password to in the connection string?

Reply:
monitor and data nodes now provide which user needs to authenticated. user names and passwords are hard coded at this time.

connections string checks the node requires authentication, and if user password is supplied during provisioning, then creates the connection string accordingly